R/teal_data_module.R

```diff
@@ -149,6 +153,19 @@ teal_transform_module <- function(ui = function(id) NULL,
       ui = ui,
       server = function(id, data) {
         data_out <- server(id, data)
+
+        if (inherits(data_out, "reactive.event")) {
+          # This warning message partially detects when `eventReactive` is used in `data_module`.
+          warning(
+            paste(
+              "srv_teal_data",
+              "Using eventReactive in teal_transform module server code should be avoided as it",
+              "may lead to unexpected behavior. See the vignettes for more information ",
+              "(`vignette(\"data-transform-as-shiny-module\", package = \"teal\")`)."
+            )
```

Comment:
warning doesn't need `paste` as `...` is concatenated. Let's also point out that problem lays in a `teal_transform_module` and not in the internal teal server function

```suggestion
            "teal_transform_module()",
            "Using eventReactive in teal_transform module server code should be avoided as it",
            "may lead to unexpected behavior. See the vignettes for more information ",
            "(`vignette(\"data-transform-as-shiny-module\", package = \"teal\")`).",
            call. = FALSE
```